### PR TITLE
Add status and filter branch

### PR DESCRIPTION
**Pull request by bakano98**
Two new commands are added, which are status and filter.
Status is used to mark whether someone is blacklisted/favourited in ModuleMateFinder (e.g. status 1 s/blacklist).
Filter is used to filter people in ModuleMateFinder by their tags (e.g. filter friends). This is case sensitive, and can only filter by 1 tag.

Examples:
`status 1 s/blacklist` - will blacklist the person at index 1
`filter t/CS3230` - will find all persons with the tag CS3230

Automated test cases have also been added.

Please review before merging.

### CODE REVIEW

**Review by lawwm — COMMENTED** [review]
lgtm. Maybe change remark to status in variable names?

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -34,6 +34,8 @@
     public static final String VALID_EMAIL_BOB = "bob@example.com";
     public static final String VALID_ADDRESS_AMY = "Block 312, Amy Street 1";
     public static final String VALID_ADDRESS_BOB = "Block 123, Bobby Street 3";
+    public static final String VALID_REMARK_AMY = "favourite";
+    public static final String VALID_REMARK_BOB = "blacklist";
```
[the remark this answers is not in the document]
Comment (bakano98) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:38]: Good catch. Missed out on full caps'd REMARK while mass replacing within Intellij.
Will add `Person.BLACKLIST`. Thanks for the suggestion!

**Review by Btaykb — COMMENTED** [review]
LGTM, maybe for the test files might want to update the tags to be module names as reflected in [here](https://github.com/AY2122S2-CS2103T-T13-4/tp/pull/16)

**Comment by bakano98** [comment]
> LGTM, maybe for the test files might want to update the tags to be module names as reflected in [here](https://github.com/AY2122S2-CS2103T-T13-4/tp/pull/16)

I'll let the changes (for tags to module names) be written over the current files when [clearing of data](https://github.com/AY2122S2-CS2103T-T13-4/tp/pull/16) since they will overlap. Since there are no other issues, I'll merge and close this PR.

**Pull request merged**